### PR TITLE
VLAN discovery general

### PR DESCRIPTION
**Pull request by Npeca75**
LNMS start VLANs discovery with iterating trough 'VlanStaticName'

the problem is next:

if some device does not support setting VLAN names, or user does not set them, or like in ELTEX , Vlan1 does not have name and could not be set, the LNMS will simply skip discovering that VLAN

Added simple code to iterate trough received data and set generic name "Vlan_XX" if it is missing

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [ ] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by Npeca75** [comment]
lnms1

**Comment by Npeca75** [comment]
@murrant @Jellyfrog 
is this could be systematic solution ? or i need to add files per OS / separate dir?

**Comment by Npeca75** [comment]
quick grepping through test files show that these OS are affected with this defect/bug

alfo80hd.json:                    "vlan_name": "",
allied_websmart.json:                    "vlan_name": "",
ciscosb_cbs350-4x.json:                    "vlan_name": "",
ciscosb_cbs350.json:                    "vlan_name": "",
ciscosb_sx550x-24f.json:                    "vlan_name": "",
edgecos_2100-28p.json:                    "vlan_name": "",
edgecos_ecs2100-10t.json:                    "vlan_name": "",
edgecos_ecs4610-24f.json:                    "vlan_name": "",
edgecos_es3510ma.json:                    "vlan_name": "",
edgecos_es3526xa-v2.json:                    "vlan_name": "",
eltex-mes.json:                    "vlan_name": "",
ifotec_inetpoe-4gp2gf.json:                    "vlan_name": "",
ironware.json:                    "vlan_name": "",
pegasus.json:                    "vlan_name": "",
picos.json:                    "vlan_name": "",
sm-os_gc8607.json:                    "vlan_name": "",
sm-os.json:                    "vlan_name": "",
ubiquoss-pon_9500.json:                    "vlan_name": "",
vrp_5720-32x-ei.json:                    "vlan_name": "",
vrp_5720.json:                    "vlan_name": "",
vrp_5720-vrf.json:                    "vlan_name": "",
vrp_ac6605-26.json:                    "vlan_name": "",
vrp_ar169sfp.json:                    "vlan_name": "",

**Comment by murrant** [comment]
Defining a generic VLAN X name or whatever is an acceptable solution.

**Comment by Npeca75** [comment]
> Defining a generic VLAN X name or whatever is an acceptable solution.

ok, only drawback is that every test will fail, because modified data not equal to tests/data/*.json

**Comment by Jellyfrog** [comment]
You need to fix the failing tests manually once

File: includes/discovery/vlans/q-bridge-mib.inc.php [hunk 1]
```diff
@@ -37,6 +37,20 @@
         }, []);
     }
 
+    //if 'VlanStaticName' is not set, LNMS won't discover Vlan although Vlan is functional on device
```
Comment (murrant) [inline includes/discovery/vlans/q-bridge-mib.inc.php:40]: You are fixing this in the wrong spot.

File: includes/discovery/vlans/q-bridge-mib.inc.php [hunk 2]
```diff
@@ -37,6 +37,20 @@
...
+    foreach ($tagoruntag as $vId => $tmp) {
+        if (! $vlans[$vtpdomain_id]['dot1qVlanStaticName']) {
+            $vlans[$vtpdomain_id] = ['dot1qVlanStaticName' => 'Vlan_' . $vId];
+            d_echo("Vlans: assigned generic Vlan_$vId name");
+        }
+    }
+
     foreach ($vlans as $vlan_id => $vlan) {
         d_echo(" $vlan_id");
         if (is_array($vlans_db[$vtpdomain_id][$vlan_id])) {
```
Comment (murrant) [inline includes/discovery/vlans/q-bridge-mib.inc.php:56]: Remove this check and and provide the default name below.

**Review by murrant — CHANGES_REQUESTED** [review]

File: includes/discovery/vlans/q-bridge-mib.inc.php [hunk 3]
```diff
@@ -37,6 +37,20 @@
...
+        }
+    }
+    foreach ($tagoruntag as $vId => $tmp) {
+        if (! $vlans[$vtpdomain_id]['dot1qVlanStaticName']) {
+            $vlans[$vtpdomain_id] = ['dot1qVlanStaticName' => 'Vlan_' . $vId];
+            d_echo("Vlans: assigned generic Vlan_$vId name");
+        }
+    }
+
     foreach ($vlans as $vlan_id => $vlan) {
```
Comment (murrant) [inline includes/discovery/vlans/q-bridge-mib.inc.php:54]: Need to change this to loop over the other arrays instead of $vlans.

Comment (Npeca75) [reply includes/discovery/vlans/q-bridge-mib.inc.php:54]: i don't understand

    foreach ($vlans as $vlan_id => $vlan) {
will not reach missing ID because it not exist in $vlans array
... and later
    $vlan_upd['vlan_name'] = $vlan['dot1qVlanStaticName'];
again, $vlans array dot1qVlanStaticName is used

so i could not imagine how to solve this in other way
only to fill $vlans array with missing data before reach 
    foreach ($vlans as $vlan_id => $vlan) {

Comment (Npeca75) [reply includes/discovery/vlans/q-bridge-mib.inc.php:54]: DBG before: 
Array
(
    [2] => Array
        (
            [dot1qVlanStaticName] => guest
        )

    [254] => Array
        (
            [dot1qVlanStaticName] => mgmn
        )

    [518] => Array
        (
            [dot1qVlanStaticName] => mcast
        )

)

Vlans: assigned generic Vlan_1 name  

DBG after: 
Array
(
    [2] => Array
        (
            [dot1qVlanStaticName] => guest
        )

    [254] => Array
        (
            [dot1qVlanStaticName] => mgmn
        )

    [518] => Array
        (
            [dot1qVlanStaticName] => mcast
        )

    [1] => Array
        (
            [dot1qVlanStaticName] => Vlan_1
        )

)

**Comment by Npeca75** [comment]
sorry
i am messed mi branch totally
closing for now

File: includes/discovery/vlans/q-bridge-mib.inc.php (hunk 3 above)
Comment (murrant) [reply includes/discovery/vlans/q-bridge-mib.inc.php:54]: Yeah, the VLAN code is very hard to follow.  Unfortunately, your fix makes it harder to follow.  

The easiest thing would be to combine all the data into one array during the snmp calls then we only need one foreach loop.


```php
[
  1 => array:1 [
    "dot1qVlanStaticName" => "default",
    "dot1qVlanCurrentUntaggedPorts" => "FF C0 00 00 00 00 00 00 03 FF 80 00 00"
    "dot1qVlanCurrentEgressPorts" => "FF C0 00 00 00 00 00 00 03 FF 80 00 00"
  ]
]
```

Comment (murrant) [reply includes/discovery/vlans/q-bridge-mib.inc.php:54]: Then if dot1qVlanStaticName is not defined for that vlan id, we can then generate a name.

Comment (Npeca75) [reply includes/discovery/vlans/q-bridge-mib.inc.php:54]: ok @murrant 
i made a new q-bridge-mib as you wish, completly rewritten, using new Laravel/DB etc
but i will wait what will happen with Eltex mes21xx/23xx PR to avoid test/data/json problems
after that i will send PR
